cirq/circuits/circuit.py

```diff
@@ -1124,23 +1124,23 @@ def to_unitary_matrix(
             qubit_order: ops.QubitOrderOrList = ops.QubitOrder.DEFAULT,
             qubits_that_should_be_present: Iterable[ops.QubitId] = (),
             ignore_terminal_measurements: bool = True,
-            ext: extension.Extensions = None,
             dtype: np.dtype = np.complex128) -> np.ndarray:
         """Converts the circuit into a unitary matrix, if possible.
 
         Args:
             qubit_order: Determines how qubits are ordered when passing matrices
                 into np.kron.
-            ext: The extensions to use when attempting to cast operations into
-                CompositeOperation instances.
             qubits_that_should_be_present: Qubits that may or may not appear
                 in operations within the circuit, but that should be included
                 regardless when generating the matrix.
             ignore_terminal_measurements: When set, measurements at the end of
                 the circuit are ignored instead of causing the method to
                 fail.
-            dtype: The numpy dtype for the returned unitary. Must be a complex
-                dtype.
+            dtype: The numpy dtype for the returned unitary. Defaults to
+                np.complex128. Specifying np.complex64 will run faster at the
```

Comment:
Aside: why default to complex128?  Isn't complex64 usually sufficient for NISQ.  Also this now does not match the quantum engine simulator.

Reply:
The reason I defaulted to complex128 was because tests tend to fail at `atol=1e-8` when you use complex64.

I agree that we should probably change this. Makes it faster, doesn't give up much of relevance. Opened https://github.com/quantumlib/Cirq/issues/1030